corehq/apps/api/resources/v0_5.py

```diff
@@ -1089,7 +1089,15 @@ def to_obj(action_times):
 
     def alter_list_data_to_serialize(self, request, data):
         data['meta']['local_date_timezone'] = self.local_timezone.zone
+        params = request.GET.copy()  # Makes params mutable for creating next_url below
 
+        if data['meta']['total_count'] >= data['meta']['limit']:
+            last_object = data['objects'][-1]
+            params['cursor_local_date'] = last_object.data['local_date']
+            params['cursor_user'] = last_object.data['user']
+
+            next_url = '{}?{}'.format(request.path, params.urlencode())
+            data['meta']['next'] = next_url
```

Comment:
I ended up encoding only the cursor parameters. It seems like the limit parameter needs to be directly in the request, otherwise tastypie doesn't process it correctly for pagination.

https://github.com/dimagi/commcare-hq/pull/32922/commits/121cca310ac8196535cbc986b187f4ef4433ba48